src/catch2/meson.build

```diff
@@ -10,6 +10,10 @@ conf_data = configuration_data()
 conf_data.set('CATCH_CONFIG_DEFAULT_REPORTER', 'console')
 conf_data.set('CATCH_CONFIG_CONSOLE_WIDTH', '80')
 
+windows_seh = get_option('windows_seh')
+conf_data.set('CATCH_CONFIG_WINDOWS_SEH', windows_seh)
+conf_data.set('CATCH_CONFIG_NO_WINDOWS_SEH', not windows_seh)
+
```

Comment:
mesons `cfg_data.set` undefines a symbol when passed `false`, and defines it when passed `true`. The complete rules are available on [mesons docs](https://mesonbuild.com/Configuration.html).

Reply:
This does not seem to have the correct semantics.

The `CATCH_CONFIG_FOO` semantics are
`CATCH_CONFIG_FOO` - skip Catch2's autodetection, force enable FOO
`CATCH_CONFIG_NO_FOO` - skip Catch2's autodetection, force disable FOO

autodetection only happens if neither is defined